### PR TITLE
ADBDEV-6537 Fix -Wmissing-prototypes warning

### PR DESCRIPTION
**Pull request by dnskvlnk**
Fix -Wmissing-prototypes warning

Commit 7185bf8 introduced support for zstd memory control via callbacks, but
these callbacks neither were declared as static, nor had prototypes, which
resulted in -Wmissing-prototypes warning about them. This patch does both
things, eliminating the warning.

### CODE REVIEW

**Comment by BenderArenadata** [comment]
Allure report https://allure.adsw.io/launch/83663

File: src/backend/storage/file/buffile.c [hunk 1]
```diff
@@ -1033,13 +1033,13 @@ BufFilePledgeSequential(BufFile *buffile)
 
 #define BUFFILE_ZSTD_COMPRESSION_LEVEL 1
 
-void *
+static void *
```
Comment (dkovalev1) [inline src/backend/storage/file/buffile.c:1041]: Although its not strictly required to have a function prototype in C, some compilers do require them, regardless of being static. 
Even this file has examples of function prototypes for static functions - look at makeBufFile().
I suggest to add prototypes in the beginning of this file and protect them with a HAVE_LIBZSTD macro.

Comment (dnskvlnk) [reply src/backend/storage/file/buffile.c:1041]: Thanks for the notice, done

**Comment by BenderArenadata** [comment]
Allure report https://allure.adsw.io/launch/83698

**Review by dkovalev1 — APPROVED** [review]

**Review by red1452 — COMMENTED** [review]
What about adding line to Makefile?
```
CFLAGS := $(CFLAGS) -Werror=missing-prototypes
```

**Comment by dnskvlnk** [comment]
@red1452 I don't think that changing CFLAGS is a good idea without a necessity. If we can fix it with a small patch, we would better choose that way

**Review by red1452 — APPROVED** [review]

**Comment by BenderArenadata** [comment]
Allure report https://allure.adsw.io/launch/83834

**Pull request merged**